src/io/asyncsocket.c

```diff
@@ -199,19 +225,20 @@ static MVMAsyncTask * read_bytes(MVMThreadContext *tc, MVMOSHandle *h, MVMObject
     }
 
     /* Create async task handle. */
-    MVMROOT4(tc, queue, schedulee, h, buf_type, {
+    MVMROOT5(tc, h, queue, schedulee, buf_type, async_type, {
         task = (MVMAsyncTask *)MVM_repr_alloc_init(tc, async_type);
     });
+
     MVM_ASSIGN_REF(tc, &(task->common.header), task->body.queue, queue);
     MVM_ASSIGN_REF(tc, &(task->common.header), task->body.schedulee, schedulee);
-    task->body.ops  = &read_op_table;
     ri              = MVM_calloc(1, sizeof(ReadInfo));
     MVM_ASSIGN_REF(tc, &(task->common.header), ri->buf_type, buf_type);
     MVM_ASSIGN_REF(tc, &(task->common.header), ri->handle, h);
     task->body.data = ri;
+    task->body.ops  = &read_op_table;
 
     /* Hand the task off to the event loop. */
-    MVMROOT(tc, task, {
+    MVMROOT6(tc, h, queue, schedulee, buf_type, async_type, task, {
         MVM_io_eventloop_queue_work(tc, (MVMObject *)task);
```

Comment:
i would assume the first five arguments here are being kept alive by the caller of this function, and since the pointers don't get used after this point, they could perhaps be left unrooted here